### ISSUE TITLE
Bank for Spitfire Studio Brass Pro

### ISSUE THREAD

**Issue by rgruet** [jtackaberry/reaticulate#190]
[Spitfire-Studio_Brass_Pro.txt](https://github.com/jtackaberry/reaticulate/files/9366629/Spitfire-Studio_Brass_Pro.txt)

**Comment by jtackaberry** [comment]
Thanks for the contribution @rgruet !  Sorry for leaving this hanging for ... god, months now. :(

Seeing as Studio Brass Pro is mostly identical to non-Pro from a bank definition perspective, notwithstanding the additional instruments, do you think it could make sense to split this into two collections: one SSB non-pro and one pro which contains the pro-specific instruments?

**Comment by rgruet** [comment]
That would make perfect sense, but would oblige me as a pro user to load 2 collections instead of one, and that's not my _usual_ approach with Pro versions (I'm not claiming it's the best approach ;-), so to keep consistent with myself I'd have to modify my other existing banks as well.
Another solution is to extract a stripped version of the pro defs in a separate Spitfire Studio Brass (Std) file. This requires more work in case of an update, but has my preference for now. What do you think?

**Comment by rgruet** [comment]
It turns out that there are apparently more differences between the two editions of the library than simply additional instruments in the Pro edition (list of articulations and naming per instrument).  And because I don't own the standard edition I can't check to see if the PRO definitions match.
So I'm afraid I'll limit myself to the definitions for the PRO edition. Here is the latest version:
[Spitfire-Studio_Brass_Pro.txt](https://github.com/jtackaberry/reaticulate/files/10114695/Spitfire-Studio_Brass_Pro.txt)

**Comment by robinboncoeur** [comment]
Not to stepon any toes, but I'm actually working on the (non-Pro) version for Brass. Would you like for me to continue or would you prefer to do this yourself? Happy either way.

**Comment by rgruet** [comment]
Be my guest!
The only potential problem I see with having two independent libraries is
the possible discrepancies in program number assignment, which would affect
someone switching libraries e. g. after an upgrade to the Pro version.
This would not occur with your proposal of splitting defs in 2 parts, but
as I previously said, it turned out to not be as easy as it looked.
You could tell me the cases where you made a different program number
choice, and I would adjust my defs accordingly. In this case my defs should
stay unpublished until this is all sorted out...

**Comment by rgruet** [comment]
@robinboncoeur: Oops! Sorry I didn't pay attention, and wrongly thought
your comment was from Jason. Not that it invalidates my remarks, but I was
just assuming that Jason would be more of an authority as to pick the right
program numbers than me...
I don't plan to do the non-Pro library so please go ahead. You may try to
align on my defs or not. Let me know if you have remarks/suggestions
regarding my choice of program numbers...

**Comment by robinboncoeur** [comment]
@rgruet : Sorry, Robin here. I'm actually finding your program numbers allign perfectly with what the library itself appears to indicate. I'm pretty sure that with the exception of having definitions for instruments not in their library, a user could use the Pro version of this reabank (yours) for the Studio library without a hitch. I've just resubmitted my Woodwinds reabank modified to address some areas lacking in the original, and as I address the Pro library I find there are pretty much no differences between Studio and Studio Pro. I'll be tackling the Strings next, although I understand that's already been done. Worthwhile excercise.

**Comment by rgruet** [comment]
@robinboncoeur: It's true that when writing definitions for libraries like Spitfire's using UACC codes to trigger articulations, we tend to use the UACC codes as PC numbers, so it's unlikely (or at least less likely) that defs from different authors will differ on that aspect :-)

BTW I have my own defs for Spitfire's Studio Woodwinds Pro:
[Spitfire-Studio_Woodwinds_Pro.txt](https://github.com/jtackaberry/reaticulate/files/10132872/Spitfire-Studio_Woodwinds_Pro.txt)

**Comment by robinboncoeur** [comment]
> BTW I have my own defs for Spitfire's Studio Woodwinds Pro: [Spitfire-Studio_Woodwinds_Pro.txt](https://github.com/jtackaberry/reaticulate/files/10132872/Spitfire-Studio_Woodwinds_Pro.txt)

@rgruet: thanks for that - hope tack doesn't mind us discussing this here... I notice that you sort-of stack what I call "bank headers" - the bit above and including the line that starts with the word Bank - above a single articulation definition, as they all share that same definition and just making them useful with the clone= attribute. Is this what I'm seeing?

Also, in my keen-ness to get the fixed Studio Woodwinds reabank out, I sort-of left out the legato set, failing to see a use-case for it. Well, actually it would be a good thing, I suppose. I'm beginning to wonder about the length of the dropdown when selecting in Reaticulate... this is getting bigger than BenHur.

**Comment by rgruet** [comment]
@robinboncoeur: I'm using the Reaticulate's  **clone** directive when a bank has the exact same set of definitions as another bank, or shares a part of the defs with it. That saves space and some sweat (although I have to determine the common part).

I think defs should cover **all** the articulations in a library, not only the ones I think I'm going to use... at least if i consider publishing them one day ;-)
The list of articulations in the UI doesn't look so BenHur-ish on my 1440p display. You can **adjust the scale** of Reaticulate's UI with Ctrl+Mouse wheel.

**Comment by robinboncoeur** [comment]
@rgruet: I agree with "defs should cover all the articulations in a library". I'm pretty sure all defs are covered in my list, but it would be prudent to add the legato group with clone= to the core (which is a fairly comprehensive listing) techniques. I do appreciate your input (even if it means going back for further refinements).
